astroid/cache.py

```diff
@@ -0,0 +1,100 @@
+""" Caching utilities used in various places."""
+
+from collections import OrderedDict
+from weakref import WeakSet
+
+import wrapt
+
+LRU_CACHE_CAPACITY = 128
+
+
+class LRUCache:
```

Comment:
Could you add typing to all methods of this class? We're trying to turn on `mypy` for `astroid` and adding typing as we're adding new functionality would be very beneficial in the future.
The best would be to run `mypy --strict` over this file to fix any potential typing related issues in the future.

Reply:
I've added typing to all methods. I've left `@lru_cache_astroid` and `@cached_generator` untyped for now because `@wrapt.decorator` is untyped.